### ISSUE TITLE
VoiceOver on iOS does not communicate content in container overlay

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#1913]
# Bug Report

## Description
When a container overlay is opened, it may contain content within it. VO on iOS does not read this content when the overlay is displayed, nor does it pick up that this is an element that can be interacted with. This is not an issue with the full screen overlay.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-core/#/components/terra-overlay/overlay/overlay
2. Open the container overlay example

## Expected Behavior
VoiceOver on iOS reads the container overlay content when its displayed in overlay.

## Environment

* Component Name and Version: terra-overlay
* Browser Name and Version: Safair Mobile 12 with VoiceOver on iOS 12

**Comment by bjankord** [comment]
Talked with @emilyrohrbough and @jmsv6d about this. We want to keep [this code](https://github.com/cerner/terra-core/blob/master/packages/terra-overlay/src/Overlay.jsx#L90-L107). It is used to disable focusing on children in the container that is being overlayed. 

It looks like right after [this line](https://github.com/cerner/terra-core/blob/master/packages/terra-overlay/src/Overlay.jsx#L83), we need to shift focus to the overlay.

**Comment by bjankord** [comment]
We need to explore making it optional to shift focus to relative container based overlays. In some cases, in makes sense, in others it may not. Maybe you have multiple on the page, maybe they only appear for a brief moment while loaded content, e.g. infinite scroll component. In these cases users likely don't want the focus to be moved around, but there are also scenarios it may make sense to move focus to a relative container based overlay. 

To handle this, I propose adding a callback function that will execute when a relative container based overlay is about to open, maybe we expand this to just and overlay (relative container and full screen) which would allow users to control side effects like shifting focus if it made sense for their app.

**Comment by bjankord** [comment]
Resolved in #2249
